apihub/subscription/models.py

```diff
@@ -15,19 +67,27 @@ class Subscription(Base):
     __tablename__ = "subscriptions"
 
     id = Column(Integer, primary_key=True, index=True)
-    application = Column(String, unique=True, index=True, nullable=False)
+    # duplicate tier to make it easier to query.
     tier = Column(Enum(SubscriptionTier), default=SubscriptionTier.TRIAL)
     active = Column(Boolean, default=True)
-    credit = Column(Integer, default=0)
+    # credit = Column(Integer, default=0)
     balance = Column(Integer, default=0)
     starts_at = Column(DateTime, default=datetime.now())
     expires_at = Column(DateTime)
     recurring = Column(Boolean, default=False)
     created_at = Column(DateTime, default=datetime.now())
     created_by = Column(String)
     notes = Column(String)
+
     username = Column(String, ForeignKey("users.username"), nullable=False)
     user = relationship("User", back_populates="subscriptions")
 
+    application = Column(
```

Comment:
we can reference Application model directly, then we can do Application.subscriptions

Reply:
I think it's better to keep it associated with the model than keeping it just a string column, for example it will automatically  invalidate any unregistered application if we tried to create unsubscribed application.

Reply:
agree. what I mean is that it can reference the Application model instead of Pricing model, which makes more sense to me logically.

Reply:
What makes more sense for me is that the models relation will be as following :
Each application will have different tiers, each user can subscribe for one of these tiers.
Application > SubscriptionPricing > Subscription

Reply:
Imaging this, I am getting a list of Applications, and I want to see all subscriptions from an Application, maybe some aggregated statistics in a dashboard. How would you do it?

Reply:
you are right.